### PR TITLE
Replaced Swing with JavaFX

### LINKED ISSUE #125: forwarding messages sets the forwarded message in the wrong chat: the chat of the current recipient
If one forwards a message to a user whose chat he is not currently in (normal case), the message will be in the correct chat for the recipient, however the sender will see the message in the current chat where the original message is located, and not in the chat of the recipient.

Additionally, the chat of the recipient could be opened as a shortcut/ in order to verify the correct sending of the message.

### LINKED ISSUE #126: The user currently has the option to override default themes, even though that will be reset by restart
when a user overrides the default themes, they will appear changed for the rest of the session. However, once Envoy is restarted, they are loaded back to default values.

Thus, the easiest solution would be to disable this possibility.

### LINKED ISSUE #130: JavaFX LoginDialog
Implement a dialog similar to the current `LoginDialog` using JavaFX.

### LINKED ISSUE #131: JavaFX Settings Scene
Translate the current `Settings` UI into a JavaFX `Scene`.

Because `SettingsItem` does generate a Swing component through which the user can change the setting, this will have to be adjusted.

### PR DESCRIPTION
**Pull request by CyB3RC0nN0R**
JavaFX Components:
Closes #130
Closes #131

Old Swing Bugs:
Closes #125
Closes #126

### CODE REVIEW

**Comment by CyB3RC0nN0R** [comment]
Before the branch is ready to merge, we have to

- [ ] Fix offline mode

- [ ] Add Themes

- [ ] Finish Settings

- [ ] Integrate groups

File: src/main/java/envoy/client/ui/controller/LoginScene.java [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+		attemptOfflineMode(new LoginCredentials(userTextField.getText(), passwordField.getText().toCharArray(), false));
+	}
+
+	@FXML
+	private void registerCheckboxChanged() {
+
+		// Make repeat password field and label visible / invisible
+		repeatPasswordField.setVisible(registerCheckBox.isSelected());
+		repeatPasswordLabel.setVisible(registerCheckBox.isSelected());
+
```
Comment (delvh) [inline src/main/java/envoy/client/ui/controller/LoginScene.java:123]: - [ ] Please delete blank, unnecessary line

File: src/main/java/envoy/client/ui/controller/package-info.java [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+ * Contains JavaFX scene controllers.
+ * <p>
+ * Project: <strong>envoy-client</strong><br>
+ * File: <strong>package-info.java</strong><br>
+ * Created: <strong>08.06.2020</strong><br>
+ * 
+ * @author Kai S. K. Engelbart
+ * @since Envoy Client v0.1-beta
+ */
+package envoy.client.ui.controller;
```
Comment (delvh) [inline src/main/java/envoy/client/ui/controller/package-info.java:11]: - [ ] please add newline at eof

File: src/main/resources/css/dark.css [hunk 3]
```diff
@@ -0,0 +1,4 @@
+.button{
+	-fx-background-color: rgb(105,0,153);
+	-fx-text-fill: white;
+}
```
Comment (delvh) [inline src/main/resources/css/dark.css:4]: - [ ] please add newline at EOF

File: src/main/resources/css/light.css [hunk 4]
```diff
@@ -0,0 +1,3 @@
+.button{
+	-fx-background-color: snow;
+}
```
Comment (delvh) [inline src/main/resources/css/light.css:3]: - [ ] please add newline at EOF

File: src/main/java/envoy/client/ui/IconUtil.java [hunk 5]
```diff
@@ -1,12 +1,10 @@
 package envoy.client.ui;
 
-import java.awt.Image;
 import java.io.IOException;
 import java.util.EnumMap;
 import java.util.EnumSet;
 
-import javax.imageio.ImageIO;
-import javax.swing.ImageIcon;
+import javafx.scene.image.Image;
```
Comment (DieGurke) [inline src/main/java/envoy/client/ui/IconUtil.java:8]: Spare line

**Pull request merged**